### PR TITLE
Users/singankit/log metrics update instance results

### PR DESCRIPTION
**Pull request by singankit**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py [hunk 1]
```diff
@@ -72,10 +74,10 @@ def _validate_and_load_data(target, data, evaluators, output_path, tracking_uri,
 
 
 def _validate_columns(
-    df: pd.DataFrame,
-    evaluators: Dict[str, Any],
-    target: Optional[Callable],
-    evaluator_config: Dict[str, Dict[str, str]],
+        df: pd.DataFrame,
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:77]: fix the indent

File: src/promptflow-evals/tests/evals/conftest.py [hunk 2]
```diff
@@ -115,6 +118,33 @@ def project_scope() -> dict:
...
+
+    subscription_id = project_scope["subscription_id"]
+    resource_group_name = project_scope["resource_group_name"]
+    workspace_name = project_scope["project_name"]
+
+    trace_destination = (
+        f"azureml://subscriptions/{subscription_id}/resourceGroups/{resource_group_name}/"
+        f"providers/Microsoft.MachineLearningServices/workspaces/{workspace_name}"
+    )
+    with patch("promptflow._sdk._configuration.Configuration.get_trace_destination", return_value=trace_destination):
```
Comment (ninghu) [inline src/promptflow-evals/tests/evals/conftest.py:133]: For user, how would they set this trace destination?

File: src/promptflow-evals/promptflow/evals/evaluate/_utils.py [hunk 3]
```diff
@@ -1,9 +1,151 @@
...
+    tracking_uri = _get_mlflow_tracking_uri(trace_destination=trace_destination)
+
+    # Adding line_number as index column this is needed by UI to form link to individual instance run
+    instance_results["line_number"] = instance_results.index
+
+    if run is None:
+        mlflow.set_tracking_uri(tracking_uri)
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            with mlflow.start_run(run_name=evaluation_name) as run:
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluate/_utils.py:106]: What happened if evaluation_name is None?

Comment (ninghu) [reply src/promptflow-evals/promptflow/evals/evaluate/_utils.py:106]: Synced offline. If it's None, the service will generate evaluation name.

**Review by nagkumar91 — APPROVED** [review]

**Review by luigiw — APPROVED** [review]

**Review by nagkumar91 — DISMISSED** [review]

File: src/promptflow-evals/tests/evals/conftest.py (hunk 2 above)
Comment (singankit) [reply src/promptflow-evals/tests/evals/conftest.py:133]: Had a offline chat this is temporary until a bug is fixed

File: src/promptflow-evals/promptflow/evals/_user_agent.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+from promptflow.evals._version import VERSION
+
+USER_AGENT = "{}/{}".format("promptflow-evals", VERSION)
```
Comment (luigiw) [inline src/promptflow-evals/promptflow/evals/_user_agent.py:6]: As a side note, we should send user agent to RAI service too if we don't do that today.

Comment (singankit) [reply src/promptflow-evals/promptflow/evals/_user_agent.py:6]: Will follow up on a separate PR since there is a separate PR for an associated work.

**Pull request closed without merging**